docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,480 @@
+- [x] As a system runtime administrator, the runtime should not need to know about all possible types of hardware.
+- [ ] Perhaps hooks could be added for users to inject their own image selection criteria on a given host, or annotations could be injected `TOOL RESPONSIBILITY`.
+
+### Deployment Engineer
+
+- [ ] As a deployment engineer, I want to parse an image index and find the “optimal” image for the cluster node I am aiming to run the image on `AGNOSTIC`.
+That includes being able to:
+  - Discover the image that fits the selected host.
+  - Find the best match from the nodes and images I have available.
+  - Determine that the image is not fitting the selected host.
+- [x] As a deployment engineer, I want the image compatibility check to be performed without downloading or executing the referenced image layers.
+- [x] As a deployment engineer, I should be able to add the compatibility to images already being used in production, especially for the images released before image compatibility wg was created.
+- [x] As a deployment engineer, I want to be able to specify the version and variant of an application or other user specific configuration (e.g. MPI), and not only hardware/kernel details in the compatibility specification.
+- [ ] As a deployment engineer, I want my compatibility spec runtime to be able to select the best possible runtime available on a node (e.g. runc vs nvidia vs wasm) `TOOL RESPONSIBILITY`
+- [x] As a deployment engineer, I want to be able to rank the images in a multi-platform image so that the runtime can know which one to choose when more than one image is compatible with the runtime environment.
+- [x] As a deployment engineer, I want to reuse community projects so that I don't duplicate and integrate the functionality of the already existing tools.
+
+### Registry Maintainer
+
+- [x] As a registry user or operator I want to have a common way of inspecting information about image compatibility to enable users to find an image that best matches their system.
+- [x] As a registry operator (or user that has no control over their registry implementation), I want any compatibility changes to not depend on registry server changes or upgrades.
+
+### OCI Specification Maintainer
+
+- [x] As a spec maintainer, I want the solution to avoid breaking other specs (confidential images, image signing, existing implementations for runtimes picking images).
+- [x] As a spec maintainer, I don't want the spec to update for new hardware devices, kernel releases, or other external dependencies.
+- [x] As a spec maintainer, I don't want to overlap significantly with solutions from other specs (like SBOMs).
+
+### Security Administrator
+
+- [x] As a security administrator, I want predictable behavior from runtimes, which does not change based on unsigned content.
```

Comment:
Does this require the compatibility artifact to also be signed?

Reply:
What is the runtime behavior if the image is signed, but the compatibility artifact is not?

Reply:
If the two are pushed from the same build CI I think this case would be unlikely. But if it happened, likely the runtime would not use it. Thankfully in HPC land we rarely do proper signing and checking of things, it's an ideal more than anything else.

Reply:
You got it!